libs/image-utils/src/image_data.ts

```diff
@@ -355,3 +355,39 @@ export function toRgba(
     channelCount: getImageChannelCount(imageData),
   });
 }
+
+/**
+ * Converts an image to grayscale.
+ */
+export function toGrayscale(
+  imageData: ImageData
+): Result<ImageData, ImageProcessingError> {
+  if (isGrayscale(imageData)) {
+    return ok(imageData);
+  }
+
+  if (!isRgba(imageData)) {
+    return err({
+      kind: ImageProcessingErrorKind.UnsupportedChannelCount,
+      channelCount: getImageChannelCount(imageData),
+    });
+  }
+
+  const dst = new Uint8ClampedArray(imageData.width * imageData.height);
+  const output = createImageData(dst, imageData.width, imageData.height);
+  const input32 = new Int32Array(imageData.data.buffer);
+
+  for (let offset = 0, size = input32.length; offset < size; offset += 1) {
+    const px = input32[offset] as number;
+    const r = px & 0xff;
+    const g = (px >>> 8) & 0xff;
+    const b = (px >>> 16) & 0xff;
+    const a = (px >>> 24) & 0xff;
+
+    // Luminosity grayscale formula.
+    const luminosity = (0.21 * r + 0.72 * g + 0.07 * b) | 0;
```

Comment:
I agree we should not use gamma correction for the performance hit, although I am curious if it would make it better. I stubbed it out but haven't tried it yet:
https://github.com/votingworks/vxsuite-print-scan/blob/cf803658f913c63cf925d2633754f09ba7e6bfe8/services/print-scan/src/printer_scanner/printing.ts#L18-L56